explainer.md

```diff
@@ -1,624 +1,411 @@
 
-The SPA code requires the addition of script APIs which provide the equivalent
-of "navigation" and "ready for first render" events referenced in step 1 and 4
-of the design. The rest of the code is identical between MPA and SPA.
+Also, data can be provided via `setData`. This can be anything structured-clonable, and will be made available to Page-B.
+
+- Open question: Is this a tight-coupling with app-history, or could it be usable without it?
+- Open question: Do we need `offerElement`? The same thing could be done by adding the CSS properties.
 
 ```js
-function handleTransition() {
-  document.documentTransition.prepare(async () => {
-    await loadNextPage();
+// In Page-B
+document.performTransition((transition) => {
+  console.log(transition.data);
+  const pageAHeader = transition.offeredElements.get('header');
+  const pageBHeader = transition.createTransitionElement(document.querySelector('.header'), {
+    capture: 'container-and-child',
   });
+  transition.matchElements(pageAHeader, pageBHeader);
+});
+```
+
+This sketch is particularly half-baked. A more concrete proposal will be possible when more of the concepts are decided.
+
+- Open question: Do we need `createTransitionElement`? It could be done via adding CSS properties, but it might be clumsy if the developer is going to immediately remove the properties afterwards.
+- Open question: How does the outgoing page offer just the root for transition?
+- Open question: When is the `performTransition` callback called?
+
+## Defining the animation
+
+How will Page-B define the animation?
+
+### Default animation
+
+The potential default animations setup for the different cases are as follows. Note all of these can be overridden by the developer:
+
+- If an element exists in Page-A only (exit animation), a fade animation takes its container from opacity 1 to 0.
+- If an element exists in Page-B only (entry animation), a fade animation takes its container from opacity 0 to 1.
+- If an element exists in both Page-A and Page-B, and both are 'container and content', an animation takes the container from Page-A styles to Page-B styles (which will include the transform used for positioning), while cross-fading the two images.
+- If an element exists in both Page-A and Page-B, and neither are 'container and child', a transform animation takes its container from Page-A size/transform to Page-B size/transform, while cross-fading the two images.
+
+- Open question: What if the Page-A element is 'container and child' but the Page-B element is 'single texture'?
+
+Because the textures are sized to 100% of the container, the textures will also change size throughout the transition. How these are scaled can be changed using regular CSS features like `object-fit`.
```

Comment:
```suggestion
Because the images are sized to 100% of the container, the images will also change size throughout the transition. How these are scaled can be changed using regular CSS features like `object-fit`.
```